app/channel.go

```diff
@@ -172,6 +172,28 @@ func CreateDirectChannel(userId string, otherUserId string) (*model.Channel, *mo
 	}
 }
 
+func GetChannelUnread(channelId string) (*model.ChannelUnread, *model.AppError) {
+	result := <-Srv.Store.Channel().GetChannelUnread(channelId);
+	if result.Err != nil {
+		return nil, result.Err
+	}
+
+	channel := result.Data.(*model.ChannelUnread)
+	var channelUnread = &model.ChannelUnread{
+		TotalMsgCount: channel.TotalMsgCount,
+		MsgCount:      0,
+		MentionCount:  channel.MentionCount,
+		TeamId:        channel.TeamId,
+		NotifyProps:   channel.NotifyProps,
+	}
```

Comment:
You don't need this, you're just copying directly from one struct to another struct